torch/optim/adam.py

```diff
@@ -99,7 +100,7 @@ def _init_group(
                     # This is because kernel launches are costly on CUDA and XLA.
                     state['step'] = (
                         torch.zeros((), dtype=torch.float, device=p.device)
-                        if group['capturable'] or group['fused']
+                        if group['capturable'] or group['fused'] or is_compiling
```

Comment:
Could you explain again by this cannot be the following?  Adding another argument in the _init_group and making adding is_compiling to this code is very unideal.
```suggestion
                        if group['capturable'] or group['fused'] or torch._utils.is_compiling()
```

Reply:
Also, this will run into a similar problem for the special case of https://github.com/pytorch/pytorch/issues/103256. We can avoid this problem by asking the user to turn capturable=True when using PT2 and adding a check to verify that capturable=True when using torch.compile. We can discuss this in more detail on Tuesday.

Reply:
Hmm yeah it probably will. I'd prefer not to have a user have to set a flag, since this should be fully automatic. I could have dynamo set a hidden flag on the optimizer which is checked here, and also checked during serde if that's acceptable?

Reply:
FYI removed this and added an internal flag which we can use as we see fit